docs/content/maintenance/small-files.md

```diff
@@ -0,0 +1,316 @@
+snapshot-1
+
+./T/manifest:
+manifest-96739ac2-5e79-4978-a3bc-86c25f1a303f-1	 # delta manifest list for snapshot-2
+manifest-96739ac2-5e79-4978-a3bc-86c25f1a303f-0  # base manifest list for snapshot-2	
+manifest-f1267033-e246-4470-a54c-5c27fdbdd074-0	 # manifest file for snapshot-2
+
+manifest-09184ccc-c07f-4090-958c-cfe3ce3889e5-1	 # delta manifest list for snapshot-1 
+manifest-09184ccc-c07f-4090-958c-cfe3ce3889e5-0  # base manifest list for snapshot-1
+manifest-2b833ea4-d7dc-4de0-ae0d-ad76eced75cc-0  # manifest file for snapshot-1
+
+./T/dt=20230501/bucket-0:
+data-b75b7381-7c8b-430f-b7e5-a204cb65843c-0.orc
+
+...
+# each partition has the data written to bucket-0
+...
+
+./T/schema:
+schema-0
+```
+The new file layout as of snapshot-2 looks like
+{{< img src="/img/small-file-1.png">}}
+
+## Delete Records From Paimon Table
+
+Now let's delete records that meet the condition `dt>=20230503`. 
+In Flink SQL, execute the following statement:
+
+```sql
+DELETE FROM T WHERE dt >= '20230503';
+```
+The third `commit` takes place and it gives use `snapshot-3`. Now, listing the files 
+under the table and your will find out no partition is dropped. Instead, a new data 
+file is created for partition `20230503` to `20230510`:
+
+```bash
+./T/dt=20230510/bucket-0:
+data-b93f468c-b56f-4a93-adc4-b250b3aa3462-0.orc # newer data file created by the delete statement 
+data-0fcacc70-a0cb-4976-8c88-73e92769a762-0.orc # older data file created by the insert statement
+```
+
+This make sense since we insert a record in the second commit (represented by 
+`+I[10, 10010, 'varchar00010', '20230510']`) and then delete
+the record in the third commit. Executing `SELECT * FROM T` will return 2 rows, namely: 
+```
++I[1, 10001, 'varchar00001', '20230501']
++I[2, 10002, 'varchar00002', '20230502']
+```
+
+The new file layout as of snapshot-3 looks like
+{{< img src="/img/small-file-2.png">}}
+
+Note that `manifest-3-0` contains 8 manifest entries of `ADD` operation type, 
+corresponding to 8 newly written data files. 
+
+## Alter Paimon Table
+As you may have noticed, the number of small files will augment over successive 
+snapshots, which may lead to decreased read performance. Therefore, a full-compaction 
+is needed in order to reduce the number of small files.
+
+Execute the following statement to configure full-compaction:
+```sql
+ALTER TABLE T SET ('full-compaction.delta-commits' = '1');
+```
+
+It will create a new schema for Paimon table, namely `schema-1`, but no snapshot 
+has actually used this schema yet until the next commit.
+
+This configuration will ensure that partitions are full compacted before writing 
+ends, and since we haven't done any compaction yet, the next commit will produce 
+two snapshots, one for data written and one for full-compaction. However, we will 
+not use this configuration since Flink does not support running compaction in Flink SQL.
+
+Let's trigger the full-compaction now. Make sure you have set execution mode to `batch` 
```

Comment:
A new title?
`## Compact Table`
We can move this `Compact Table` before `Alter Table`.